### ISSUE TITLE
Silent failure with missing namespace declaration on inner viewhelper

### ISSUE THREAD

**Issue by naturefund-falk** [TYPO3/Fluid#418]
This is somehow related to this issue: https://github.com/TYPO3/Fluid/issues/8

If you use nested viewhelpers like this:
`{f:uri.page(pageUid: '{foo:bar(...}', absolute: 1)}`

If you miss to define the namespace for the inner viewhelper (foo:), the viewhelper bar is not executed and no error message is shown.
The expected behavior is an error message like the one when i miss to define the namespace of the outer viewhelper (f:uri.page).

**Comment by NamelessCoder** [comment]
Are you certain that the code is valid? In the example code you pasted, a closing parenthesis is lacking:

> {f:uri.page(pageUid: '{foo:bar(...)}', absolute: 1)}

The problem might be that your VH is not detected as an inline accessor which means the parsed result becomes a string which in turn is cast to an integer because that's the type of the argument.

It could also be a quote escaping issue which causes your code to not be parsed correctly.

You can test this by moving the expression out from arguments into the template directly and confirm the output.

**Comment by naturefund-falk** [comment]
I dont see any missing parenthesis. In the real code the single quotes are escaped, this got lost in copy and paste. But the code is working, because if a add the missing namespace declaration, the code is working.
The problem here is not in the given code, instead in a missing namespace declaration for the inner viewhelper (foo:). And because you get no error or warning message about missing namespace, you have a very hard to find bug.
Its not that fluid is not working, its about that i would expect an error message about the missing namespace, the same as i get if i forget to define the namespace of the outer viewhelper (f: in this case)
I found this bug in the typo3 8.7 LTS version, dont know which version of Fluid this version is using.

**Comment by sascha-egerer** [comment]
@naturefund-falk 
> I dont see any missing parenthesis
```
                            missing }
                                  v
{f:uri.page(pageUid: '{foo:bar(...}', absolute: 1)}
```

**Comment by NamelessCoder** [comment]
Confirmed, caused by the way the template namespace extraction pre-processor interacts with the template. The template processor is now responsible for throwing the exceptions about unhandled namespaces - but it does not scan through arguments.

I've researched and found that moving the exception throwing to TemplateParser solves the problem and in addition provides the template chunk as part of the error message - so it seems like that's the way to solve this one. I added a check if namespace is valid in `initializeViewHelperAndAddItToStack`:

```php
        if (!$viewHelperResolver->isNamespaceValid($namespaceIdentifier)) {
            throw $this->createParsingRelatedExceptionWithContext(new UnknownNamespaceException('Unknown Namespace: ' . $namespaceIdentifier), $state->getIdentifier());
        }
```

This can be done independently of removing the exception throwing from the template pre-processor, but removing it from there will at the same time improve the quality of the error message and removes a semi-redundant `preg_match` and iteration.
